### PR TITLE
Add iam based access for dynamodb

### PR DESCRIPTION
**Pull request by chaitanyasingla-dt**
This pull request introduces IAM-based access for DynamoDB functionality. The AWS SDK will now automatically retrieve the secret access key and access key ID from the appropriate chain provider based on the specified AWS region. This retrieval is for the assumed role attached to the service's instance.

cc: @Sachin4403

### CODE REVIEW

File: store/dynamodb/db.go [hunk 1]
```diff
@@ -69,6 +72,9 @@ type Config struct {
 	// dual stack (IPv4 and IPv6).
 	// (Optional) Defaults to False.
 	DisableDualStack bool
+
+	// If iamBasedAccessEnabled is enabled, accessKey and secretKey will be fetched using IAM temporary credentials
+	IamBasedAccessEnabled bool
```
Comment (Sachin4403) [inline store/dynamodb/db.go:77]: Rename  this flag to roleBasedAccess.

File: store/dynamodb/db.go [hunk 2]
```diff
@@ -111,6 +139,20 @@ func NewDynamoDB(config Config, measures metric.Measures) (store.S, error) {
 	}, nil
 }
 
+func getAwsRegionForCassandra(config Config) (string, error) {
```
Comment (Sachin4403) [inline store/dynamodb/db.go:142]: Please change this function name.

**Review by Sachin4403 — CHANGES_REQUESTED** [review]

File: store/dynamodb/db.go (hunk 1 above)
Comment (chaitanyasingla-dt) [reply store/dynamodb/db.go]: Done

File: store/dynamodb/db.go (hunk 2 above)
Comment (chaitanyasingla-dt) [reply store/dynamodb/db.go]: Done

**Review by Sachin4403 — APPROVED** [review]

**Comment by Sachin4403** [comment]
@schmidtw 
Please review and merge this pull request.

**Review by Sachin4403 — APPROVED** [review]

**Pull request closed without merging**